redwood-treehouse-host/src/commonMain/kotlin/app/cash/redwood/treehouse/TreehouseAppContent.kt

```diff
@@ -365,16 +368,30 @@ private class ViewContentCodeBinding<A : AppService>(
       val scopedAppService = session.appService.withScope(ziplineScope)
       val treehouseUi = contentSource.get(scopedAppService)
       treehouseUiOrNull = treehouseUi
+      val restoredId = viewOrNull?.restoredId
+      val restoredState = if (restoredId != null) app.stateStore.get(restoredId) else null
```

Comment:
There’s lots of concurrency hazards here . . .

 * `get()` is called on a different dispatcher from `put()`
 * if `get()` suspends for a long time, it’s unclear what happens to the UI?
 * what happens if we cancel while waiting for get() to return?

I think this is only solved by tests unfortunately. (No action to take here.)

Reply:
Well, maybe we should lock down which dispatcher calls happen on